docs/JAX_for_PyTorch_users.md

```diff
@@ -0,0 +1,890 @@
+Similarly to PyTorch tensors, JAX Array attributes describe the array's shape, dtype and device:
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+id: 4AwolCS1OHw2
+outputId: 3defd0bb-4548-4afa-fe9c-3e84b446df24
+---
+x_jax = jnp.ones((3, 4))
+print(f"Shape of tensor: {x_jax.shape}")
+print(f"Datatype of tensor: {x_jax.dtype}")
+print(f"Device tensor is stored on: {x_jax.device}")
+```
+
++++ {"id": "S4CxmHSaKz-r"}
+
+However, there are some notable differences between PyTorch tensors and JAX Arrays:
+- JAX Arrays are immutable
+- The default integer and float dtypes are int32 and float32
+- The default device corresponds to the available accelerator, e.g. cuda:0 if one or multiple GPUs are available.
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+id: cDVk9NjNKqJ2
+outputId: 3769b2fe-c241-447a-a9d9-c622b568e963
+---
+try:
+  x_jax[0, 0] = 100.0
+except TypeError as e:
+  print(e)
+
+
+x_torch = torch.tensor([1, 2, 3, 4])
+x_jax = jnp.array([1, 2, 3, 4])
+print(f"Default integer dtypes, PyTorch: {x_torch.dtype} and Jax: {x_jax.dtype}")
+
+x_torch = torch.zeros(3, 4)
+x_jax = jnp.zeros((3, 4))
+print(f"Default float dtypes, PyTorch: {x_torch.dtype} and Jax: {x_jax.dtype}")
+print(f"Default devices, PyTorch: {x_torch.device} and Jax: {x_jax.device}")
+```
+
++++ {"id": "1o1u2N1lL7I3"}
+
+Note that this tutorial does not cover the API to set values on indices on an existing JAX Array. For more details, please check the corresponding JAX documentation: https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ndarray.at.html.
+
++++ {"id": "_x7LutxoC3Eq"}
+
+### Devices and accelerators
+
+Using the PyTorch API, we can check whether we have GPU accelerators available with `torch.cuda.is_available()`. In JAX, we can check available devices as follows:
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+id: s0k84UEMQSwL
+outputId: 71b3658b-45ee-4fe1-a972-986f2e0da950
+---
+print(f"Available devices given a backend (gpu or tpu or cpu): {jax.devices()}")
+# Define CPU and CUDA devices
+cpu_device = jax.devices("cpu")[0]
+cuda_device = jax.devices("cuda")[0]
+print(cpu_device, cuda_device)
+```
+
++++ {"id": "u4zGIZt9RyDF"}
+
+Let's briefly explore how to create arrays on CPU and CUDA devices.
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+id: 9ZSyG8Q6Q5Jw
+outputId: 68da1b2b-6803-4382-c851-b1c1062c60dd
+---
+# create an array on CPU and check the device
+x_cpu = jnp.ones((3, 4), device=cpu_device)
+print(x_cpu.device, )
+# create an array on GPU
+x_gpu = jnp.ones((3, 4), device=cuda_device)
+print(x_gpu.device)
+```
+
+In JAX it is possible to implicitly move an array from one device to another when we operate with arrays on different devices. Namely, when we create a JAX Array **without explicitly** specifying a device, unlike PyTorch, in the computation it can be implicitly moved to the explicitly specified device of another JAX Array.
```

Comment:
```suggestion
In PyTorch we are used to device placement always being explicit. JAX can operate this way via explicit device placement as above, but unless the device is specified the array will remain *uncommitted*: i.e. it will be stored on the default device, but allow implicit movement to other devices when necessary:
```